src/wp-includes/fonts/class-wp-font-library.php

```diff
@@ -0,0 +1,145 @@
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var array
+	 */
+	private $collections = array();
+
+	/**
+	 * Container for the main instance of the class.
+	 *
+	 * @since 6.5.0
+	 * @var WP_Font_Library|null
+	 */
+	private static $instance = null;
+
+	/**
+	 * Register a new font collection.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $slug         Font collection slug.
+	 * @param array  $data_or_file Font collection data array or a path/URL to a JSON file
+	 *                             containing the font collection.
+	 *                             See {@see wp_register_font_collection()} for the supported fields.
+	 * @return WP_Font_Collection|WP_Error A font collection if it was registered successfully,
+	 *                                     or WP_Error object on failure.
+	 */
+	public function register_font_collection( $slug, $data_or_file ) {
+		$new_collection = new WP_Font_Collection( $slug, $data_or_file );
+
+		if ( $this->is_collection_registered( $new_collection->slug ) ) {
+			$error_message = sprintf(
+			/* translators: %s: Font collection slug. */
+				__( 'Font collection with slug: "%s" is already registered.' ),
+				$new_collection->slug
+			);
+			_doing_it_wrong(
+				__METHOD__,
+				$error_message,
+				'6.5.0'
+			);
+			return new WP_Error( 'font_collection_registration_error', $error_message );
+		}
+		$this->collections[ $new_collection->slug ] = $new_collection;
+		return $new_collection;
+	}
+
+	/**
+	 * Unregisters a previously registered font collection.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $slug Font collection slug.
+	 * @return bool True if the font collection was unregistered successfully and false otherwise.
+	 */
+	public function unregister_font_collection( $slug ) {
+		if ( ! $this->is_collection_registered( $slug ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				/* translators: %s: Font collection slug. */
+				sprintf( __( 'Font collection "%s" not found.' ), $slug ),
+				'6.5.0'
+			);
+			return false;
+		}
+		unset( $this->collections[ $slug ] );
+		return true;
+	}
+
+	/**
+	 * Checks if a font collection is registered.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $slug Font collection slug.
+	 * @return bool True if the font collection is registered and false otherwise.
+	 */
+	private function is_collection_registered( $slug ) {
+		return array_key_exists( $slug, $this->collections );
+	}
+
+	/**
+	 * Gets all the font collections available.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return array List of font collections.
+	 */
+	public function get_font_collections() {
+		return $this->collections;
+	}
+
+	/**
+	 * Gets a font collection.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $slug Font collection slug.
+	 * @return WP_Font_Collection|WP_Error Font collection object,
+	 *                                     or WP_Error object if the font collection doesn't exist.
+	 */
+	public function get_font_collection( $slug ) {
+		if ( array_key_exists( $slug, $this->collections ) ) {
```

Comment:
Shouldn't this use `$this->is_collection_registered()` instead of repeating this code?

```suggestion
		if ( $this->is_collection_registered( $slug ) ) {
```

Reply:
Yes! Addressing in Gutenberg here: https://github.com/WordPress/gutenberg/pull/58691/commits/b51db012b8c26f48a9d402253d3cf3071f530c3a

Reply:
Changed in https://github.com/WordPress/wordpress-develop/pull/6027/commits/095fd857413b24fd674284cd3229e4100cf0d994 ✅